main.py

```diff
@@ -558,13 +562,28 @@ def main(cl_arguments):
     if args.do_full_eval:
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
+        diagnostic_tasks = []
+        target_tasks_remove = []
+        for index, task in enumerate(target_tasks):
+            if task.name in task_modules.ALL_DIAGNOSTICS:
+                diagnostic_tasks.append(task)
+                target_tasks_remove.append(index)
+        # To avoid evaluating twice on a diagnostic task.
+        for index in sorted(target_tasks_remove, reverse=True):
+            del target_tasks[index]
+        if len(diagnostic_tasks) > 0:
```

Comment:
why do you need this? If `len(diagnostic_tasks) == 0` then the for loop just terminates